.buildkite/ftr_configs.yml

```diff
@@ -491,8 +491,12 @@ enabled:
   - x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/rule_creation/trial_license_complete_tier/configs/serverless.config.ts
   - x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/rule_patch/trial_license_complete_tier/configs/ess.config.ts
   - x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/rule_patch/trial_license_complete_tier/configs/serverless.config.ts
+  - x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/rule_patch/basic_license_essentials_tier/configs/ess.config.ts
+  - x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/rule_patch/basic_license_essentials_tier/configs/serverless.config.ts
```

Comment:
Not sure I understand what these combinations mean:

- `basic_license_essentials_tier` + `ess.config.ts` (how essentials tier is related to ESS?)
- `basic_license_essentials_tier` + `serverless.config.ts` (how basic license is related to Serverless?)

Same applies to other combinations in the structure we're creating:

- `trial_license_complete_tier` + `ess.config.ts` (how complete tier is related to ESS?)
- `trial_license_complete_tier` + `serverless.config.ts` (how trial license is related to Serverless?)

I think the reason for that is that we move the tests from the legacy folder w/o doing fundamental restructuring of them around ESS licensing and Serverless tiers. Which is ok at this point. But I think the PR doesn't fully fulfil the requirements of https://github.com/elastic/kibana/issues/151877. If you look closely, it proposes to:

- [ ] Remove any duplicate tests.
- [ ] Move any features not requiring a Platinum license to the `basic` group (which should be the case for most Security Solution features).
- [ ] For features that require a Platinum license, we should test that they are not accessible with the basic license in the `basic` group, and we should test that they work correctly in the `security_and_spaces` group.

I still can see duplication in the moved tests. Sometimes we define the same test twice in two different files, and since both files are marked as `@ess @serverless`, we execute the same test 4 times (!). We should be executing most tests twice (in ESS and Serverless) and we should define them only once in the code.

See this example, where the `should patch a single rule property of name using a rule_id` test is identical and implemented twice in two files:

`security_solution_api_integration/test_suites/detections_response/rules_management/rule_patch/basic_license_essentials_tier/patch_rules.ts`:

```ts
  describe('@ess @serverless patch_rules', () => {
    describe('patch rules', () => {
      beforeEach(async () => {
        await createAlertsIndex(supertest, log);
      });

      afterEach(async () => {
        await deleteAllAlerts(supertest, log, es);
        await deleteAllRules(supertest, log);
      });

      it('should patch a single rule property of name using a rule_id', async () => {
        await createRule(supertest, log, getSimpleRule('rule-1'));

        // patch a simple rule's name
        const { body } = await supertest
          .patch(DETECTION_ENGINE_RULES_URL)
          .set('kbn-xsrf', 'true')
          .set('elastic-api-version', '2023-10-31')
          .send({ rule_id: 'rule-1', name: 'some other name' })
          .expect(200);

        const outputRule = getSimpleRuleOutput();
        outputRule.name = 'some other name';
        outputRule.revision = 1;
        const expectedRule = updateUsername(outputRule, ELASTICSEARCH_USERNAME);

        const bodyToCompare = removeServerGeneratedProperties(body);
        expect(bodyToCompare).to.eql(expectedRule);
      });
```

`security_solution_api_integration/test_suites/detections_response/rules_management/rule_patch/trial_license_complete_tier/patch_rules.ts`:

```ts
  describe('@ess @serverless @skipInQA patch_rules', () => {
    describe('patch rules', () => {
      beforeEach(async () => {
        await createAlertsIndex(supertest, log);
      });

      afterEach(async () => {
        await deleteAllAlerts(supertest, log, es);
        await deleteAllRules(supertest, log);
      });

      it('should patch a single rule property of name using a rule_id', async () => {
        await createRule(supertest, log, getSimpleRule('rule-1'));

        // patch a simple rule's name
        const { body } = await supertest
          .patch(DETECTION_ENGINE_RULES_URL)
          .set('kbn-xsrf', 'true')
          .set('elastic-api-version', '2023-10-31')
          .send({ rule_id: 'rule-1', name: 'some other name' })
          .expect(200);

        const outputRule = updateUsername(getSimpleRuleOutput(), ELASTICSEARCH_USERNAME);

        outputRule.name = 'some other name';
        outputRule.revision = 1;
        const bodyToCompare = removeServerGeneratedProperties(body);
        expect(bodyToCompare).to.eql(outputRule);
      });
```

Let's keep https://github.com/elastic/kibana/issues/151877 open for now. I'd like to try to make a follow-up attempt to restructure this around licensing and tiers in a follow-up PR.

Reply:
That's fair! I'll go ahead and keep that one open.